### PR TITLE
TRUNK-5946: ProgramAttributeType Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

switched the `ProgramAttributeType` domain from Hibernate XML mappings to annotations by removing the XML mapping file and  annotating the class with annotations

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5946

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by wikumChamith** [comment]
@ManojLL can we get the merge conflicts resolved?

**Comment by ManojLL** [comment]
@wikumChamith Resolved the merge conflict

File: api/src/main/java/org/openmrs/ProgramAttributeType.java [hunk 1]
```diff
@@ -14,8 +14,21 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "program_attribute_type")
 @Audited
 public class ProgramAttributeType extends BaseAttributeType<PatientProgram> implements AttributeType<PatientProgram> {
+
+	@Id
+	@Column(name = "program_attribute_type_id")
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
```
[the remark this answers is not in the document]
Comment (ManojLL) [reply api/src/main/java/org/openmrs/ProgramAttributeType.java]: @wikumChamith I updated the PR

**Review by wikumChamith — APPROVED** [review]
@dkayiwa can you take a look at this PR?

**Pull request merged**